index.d.ts

```diff
@@ -0,0 +1,590 @@
+/// <reference types="node" />
+
+import * as tls from "tls";
+import * as net from "net";
+
+export class Kafka {
+    constructor(config: KafkaConfig);
+    producer(config?: ProducerConfig): Producer;
+    consumer(config?: ConsumerConfig): Consumer;
+    admin(config?: AdminConfig): Admin;
+    logger(): Logger;
+}
+
+export interface KafkaConfig {
+    brokers: string[];
+    ssl?: tls.SecureContextOptions;
+    sasl?: SASLOptions;
+    clientId?: string;
+    connectionTimeout?: number;
+    authenticationTimeout?: number;
+    requestTimeout?: number;
+    enforceRequestTimeout?: boolean;
+    retry?: RetryOptions;
+    socketFactory?: ISocketFactory;
+    logLevel?: logLevel;
+    logCreator?: logCreator;
+    allowExperimentalV011?: boolean;
```

Comment:
You can remove `allowExperimentalV011`; this is enabled by default since v1.4.0. The reason it is still here is that we have a lot of useful tests checking the migration from the 0.10 format to 0.11 and we haven't decided what to do with them, but we should remove this flag from the public interface.

Reply:
removed.